test/runnable/testbtst.d

```diff
@@ -91,5 +91,68 @@ int main()
     auto d = BitArray(bd);
 
     assert(a <  d);
+}
+
+/***************************************/
+
+// https://issues.dlang.org/show_bug.cgi?id=18748
+
+int bt_32_imm(in uint* p)
+{
+    enum bitnum = 1;
+    return ((p[bitnum >> 5] & (1 << (bitnum & 31)))) != 0;
+}
+
+void test18748()
+{
+    version (linux)
+    {
+        import core.sys.posix.sys.mman;
+        import core.sys.posix.unistd;
+        // Allocate two pages.
+        immutable sz = 2 * sysconf(_SC_PAGESIZE);
+        auto m = mmap(null, sz, PROT_READ, MAP_PRIVATE | MAP_ANON, -1, 0);
+        // Discard the higher page. It becomes unreadable.
+        munmap(m + sz / 2, sz / 2);
+        // Try looking at the last 4 bytes of the readable page.
+        uint* p = cast(uint*) (m + sz / 2 - uint.sizeof);
+        bt_32_imm(p);
+        munmap(m, sz / 2); // Free the readable page.
+    }
+}
+
+/***************************************/
+
+// https://issues.dlang.org/show_bug.cgi?id=18749
+
+ulong f(ulong* p, uint shift)
+{
+    return (*p >> shift) & 1;
+}
+
+ulong g(ulong* p, ulong shift)
+{
+    return f(p, cast(uint) shift);
+}
+
+void test18749()
+{
+/+
+    enum shift = uint.max + 1L;
+    assert(cast(uint) shift == 0);
+    ulong s = 1;
+    assert(g(&s, shift));
++/
```

Comment:
What about this ?

Reply:
that's from another fix that wasn't merged yet.

Reply:
I merged it yesterday, this needs a rebase.